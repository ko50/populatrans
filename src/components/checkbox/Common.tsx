import styles from "styles/components/checkbox/common.module.scss";

export const Checkbox: React.FC<
  React.InputHTMLAttributes<HTMLInputElement>
> = ({ ...props }) => {
  return <input className={styles.checkbox} type="checkbox" {...props} />;
};
