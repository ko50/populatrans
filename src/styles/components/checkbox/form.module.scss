@use "styles/utils/colors.scss";
@use "styles/utils/fonts.scss";

.form {
  display: flex;
  align-items: center;
  color: colors.$carbon;
  font-family: fonts.$kosugi;
  cursor: pointer;

  label {
    margin-left: 0.5rem;
  }
}
