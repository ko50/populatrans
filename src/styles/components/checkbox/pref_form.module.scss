@use "styles/utils/mq.scss";

.wrapper {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  gap: 1rem;

  @include mq.mq("sm") {
    grid-template-columns: repeat(3, 1fr);
    font-size: 0.8rem;
  }
}
