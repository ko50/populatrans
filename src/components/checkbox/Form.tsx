import { Checkbox } from "components/checkbox/Common";

import styles from "styles/components/checkbox/form.module.scss";

type Props = {
  label: string;
};

export const CheckboxForm: React.FC<
  Props & React.InputHTMLAttributes<HTMLInputElement>
> = ({ label, ...props }) => {
  return (
    <div className={styles.form}>
      <Checkbox id={props.id} {...props} />
      <label htmlFor={props.id}>{label}</label>
    </div>
  );
};
