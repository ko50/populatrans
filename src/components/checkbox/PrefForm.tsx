import { CheckboxForm } from "components/checkbox/Form";
import { Prefecture } from "models/prefecture";

type Props = {
  prefList: Prefecture[];
  onSelect: (pref: Prefecture) => void;
};

export const PrefForm: React.FC<Props> = ({ prefList, onSelect }) => {
  return (
    <div>
      {prefList.map((p) => {
        return (
          <CheckboxForm
            key={p.code}
            label={p.name}
            id={p.code.toString()}
            onChange={() => onSelect(p)}
          />
        );
      })}
    </div>
  );
};
