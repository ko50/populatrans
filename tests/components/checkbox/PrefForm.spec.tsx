import React from "react";
import renderer from "react-test-renderer";
import { PrefForm } from "components/checkbox/PrefForm";

describe("prefecture form component", () => {
  it("should render correctly", () => {
    const tree = renderer
      .create(<PrefForm prefList={[]} onSelect={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render correctly with data", () => {
    const tree = renderer
      .create(
        <PrefForm
          prefList={[
            { code: 1, name: "茨城県" },
            { code: 2, name: "埼玉県" },
          ]}
          onSelect={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
