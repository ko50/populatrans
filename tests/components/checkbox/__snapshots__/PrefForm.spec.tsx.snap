// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prefecture form component should render correctly 1`] = `
<div
  className="wrapper"
/>
`;

exports[`prefecture form component should render correctly with data 1`] = `
<div
  className="wrapper"
>
  <div
    className="form"
  >
    <input
      className="checkbox"
      id="1"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="1"
    >
      茨城県
    </label>
  </div>
  <div
    className="form"
  >
    <input
      className="checkbox"
      id="2"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="2"
    >
      埼玉県
    </label>
  </div>
</div>
`;
